src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -2466,6 +2466,160 @@ public void CopyTo(Span<T> other)
                 );
         }
 
+        [Fact, WorkItem(35146, "https://github.com/dotnet/roslyn/issues/35146")]
+        public void ReadOnlyMethod_RefLikeStructParameter()
+        {
+            var csharp = @"
+using System;
+
+public ref struct S<T>
+{
+    public readonly void M(Span<T> x) { }
+
+    public unsafe static void N(S<byte> b)
+    {
+        Span<byte> x = stackalloc byte[5];
+        b.M(x);
+    }
+}
+";
+            var comp = CreateCompilationWithMscorlibAndSpan(csharp, TestOptions.UnsafeDebugDll);
+            comp.VerifyDiagnostics();
+        }
+
+        [Fact, WorkItem(35146, "https://github.com/dotnet/roslyn/issues/35146")]
+        public void ReadOnlyStruct_Property_RefLikeStruct_01()
+        {
+            var csharp = @"
+using System;
+
+public ref struct S<T>
+{
+    public Span<T> P { get => default; readonly set {} }
```

Comment:
>readonly set {} [](start = 39, length = 15)

It still bugs me that this is allowed.

Reply:
I can't really think of a case for a setter of `Span<T>`, but for non `ref structs`, it might cover an instance method which only touches some shared static state (ignoring whether or not that is good/bad API design).

Reply:
side table?